From 6a5a682d4ad5f7a92ae74da5b1bc86709844ade5 Mon Sep 17 00:00:00 2001
From: Peng Wu <wupeng58@huawei.com>
Date: Wed, 23 Feb 2022 18:45:21 +0800
Subject: [PATCH] mm: Reserve field in mm_struct for memory reliable
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6a5a682d4ad5f7a92ae74da5b1bc86709844ade5
Modified-by-SEL: Yes, refreshed due to context change


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4SK3S
DTS: #896
CVE: NA

-------------------------------------------------

Reserved a variable in mm_struct while will be used by memory reliable.

Signed-off-by: Peng Wu <wupeng58@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/mm_types.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/linux/mm_types.h
+++ b/include/linux/mm_types.h
@@ -596,6 +596,10 @@ struct mm_struct {
 #ifdef CONFIG_IOMMU_SVA
 		u32 pasid;
 #endif
+
+#ifdef CONFIG_MEMORY_RELIABLE
+		atomic_long_t reserve_0;
+#endif
 	} __randomize_layout;
 
 	KABI_RESERVE(1)
